app/src/main/java/org/oppia/android/app/options/AppLanguageSelectionViewModel.kt

```diff
@@ -20,29 +25,46 @@ class AppLanguageSelectionViewModel @Inject constructor(
   val selectedLanguage = MutableLiveData<OppiaLanguage>()
   private val appLanguageRadioButtonListener = fragment as AppLanguageRadioButtonListener
 
+  private val appLanguageResultLiveData: LiveData<AsyncResult<List<OppiaLanguage>>> by lazy {
+    translationController.getSupportedAppLanguages().toLiveData()
+  }
+
+  private val appLanguageListLiveData: LiveData<List<OppiaLanguage>> by lazy {
```

Comment:
Hi Ben used this technique to maintain consistency as it is similarly used in OptionsControlViewModel and also across other VIewModel classes loading data via controllers, what should be the alternative for this in this case. Thanks.

Reply:
You only need one process method here, not two. That also lets you remove some of the extra LiveDatas which will simplify the data pipeline.

Reply:
This doesn't seem completed yet--we shouldn't be transforming the existing LiveData. Instead, transform translationController.getSupportedAppLanguages().toLiveData() directly to avoid needing the extra field.

Also, we still have 2 process methods: processAppLanguageResult, processAppLanguageList. We really just need one.